src/vcpkg/commands.ci.cpp

```diff
@@ -173,6 +173,8 @@ namespace
                             const std::map<PackageSpec, BuildResult>& known,
                             View<std::string> parent_hashes)
     {
+        // With parent hashes, ports are merely "auto selected" unless the abi hash changed.
+        auto const default_request_type = parent_hashes.empty() ? RequestType::USER_REQUESTED : RequestType::AUTO_SELECTED;
```

Comment:
I'm also really not a fan of trying to overload `RequestType` like this because that controls a bunch of other settings, like whether default features are added:

https://github.com/microsoft/vcpkg-tool/blob/d5c355215e3bb089c04f7ef2dd16854f82919462/src/vcpkg/dependencies.cpp#L269-L272

as well as `--head`/`--editable` handling.

(I know that once we have an `ActionPlan` we are out of dependencies.cpp but that this is already being used to track other things strongly suggests that it is not the right tool to track what you're trying to track here. If all we care about is "is the abi hash in parent_hashes" let's ask that question directly at reporting time rather than trying to smuggle it through this setting)